### ISSUE TITLE
Artifact along autoshader border

### ISSUE THREAD

**Issue by TokisanGames** [TokisanGames/Terrain3D#276]
When far away, there's pixelated noise along the border between autoshaded and manually painted areas.

![image](https://github.com/TokisanGames/Terrain3D/assets/632766/eca2a3ae-2526-4690-a4cc-20916cd3e6c1)

![image](https://github.com/TokisanGames/Terrain3D/assets/632766/a1fa4c4b-785a-4ae7-a1df-b3283a6c8fb4)

Looks similar to #244 and may be related or have the same cause.

**Comment by TokisanGames** [comment]
This can be fixed by either removing the overlay shader optimization

```
if (out_mat.blend > 0.f) { 
to
if (true) {

```

Or by moving the secondary texture lookups into the main function scope, moving them before the `if out_mat.blend>0.`
```
vec4 albedo_ht2 = texture(...)
vec4 normal_rg2 = texture(...);
if (out_mat.blend > 0.f) {
```

**Comment by TokisanGames** [comment]
The above workarounds also fixes #244, so this is a duplicate. And both will be fixed with the latter as the branch does seem to still provide some optimization on my 3070.
